File: week3/wonbin/BOJ_큐.java
import java.util.*;
import java.io.*;

public class Main{
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();

        int n = Integer.parseInt(br.readLine());
        Queue<Integer> queue = new LinkedList<>();
        int last = -1;

        for(int i = 0; i < n; i++){
            String  input = br.readLine();

            if(input.startsWith("push")){
                int x = Integer.parseInt(input.split(" ")[1]);
                queue.offer(x);
                last = x;
            } else if(input.equals("pop")){
                sb.append(queue.isEmpty() ? -1 : queue.poll()).append('\n');
            } else if(input.equals("size")){
                sb.append(queue.size()).append('\n');
            } else if(input.equals("empty")){
                sb.append(queue.isEmpty() ? 1 : 0).append('\n');
            } else if(input.equals("front")){
                sb.append(queue.isEmpty() ? -1 : queue.peek()).append('\n');
            } else if(input.equals("back")){
                sb.append(queue.isEmpty() ? -1 : last).append('\n');
            }
        }
        System.out.print(sb);
    }
}

File: week3/hayeounglee/BOJ_큐.py
from collections import deque
import sys

input = sys.stdin.readline

n = int(input())
q = deque()

for _ in range(n):
    command = list(input().split())
    if command[0] == 'push':
        q.append(command[1])
    elif command[0] == 'pop':
        if len(q) == 0:
            print(-1)
        else:
            print(q.popleft())
    elif command[0] == 'front':
        if len(q) == 0:
            print(-1)
        else:
            print(q[0])
    elif command[0] == 'back':
        if len(q) == 0:
            print(-1)
        else:
            print(q[-1])
    elif command[0] == 'empty':
        print(1 if len(q) == 0 else 0)
    elif command[0] == 'size':
        print(len(q))
